python/pyspark/sql/dataframe.py

```diff
@@ -3058,7 +3058,7 @@ def toDF(self, *cols: "ColumnOrName") -> "DataFrame":
         jdf = self._jdf.toDF(self._jseq(cols))
         return DataFrame(jdf, self.sql_ctx)
 
-    def transform(self, func: Callable[["DataFrame"], "DataFrame"]) -> "DataFrame":
+    def transform(self, func: Callable[..., "DataFrame"], *args: Any, **kwargs: Any) -> "DataFrame":
```

Comment:
Note: to properly type this, we would need [PEP-612](https://www.python.org/dev/peps/pep-0612/) and python 3.10. An alternative is to provide a long list of overloads for "most" cases.